src/vm/jitinterface.cpp

```diff
@@ -7251,103 +7251,6 @@ bool getILIntrinsicImplementationForUnsafe(MethodDesc * ftn,
     return false;
 }
 
-bool getILIntrinsicImplementationForVolatile(MethodDesc * ftn,
```

Comment:
These replacements are much simpler IL that the IL has better chance of generating a good code for.

We should keep this unless you want to do the analysis to prove that this does not hurt code quality anywhere.

Reply:
(My guess is that the analysis would show that this does improve code quality in some cases quite a bit.)

Reply:
> We should keep this unless you want to do the analysis to prove that this does not hurt code quality anywhere.

I did a basic check, and it actually resulted in better code.  But I'm not sure how to prove that it won't hurt things anywhere, so I'll just close this and we can leave it not shared.

Reply:
> so I'll just close this and we can leave it not shared.

Unless you're suggesting I leave Volatile.cs in shared, and put:
```C#
#if CORECLR
[Intrinsic]
#endif
```
on each method?

Reply:
> Unless you're suggesting I leave Volatile.cs in shared, and just let the JIT in coreclr replace them anyway?

Right, that's what I meant.

Reply:
Ok

Reply:
You can mark these methods with `[Intrinsic]` as form of documentation that the JIT/VM may be replacing the body you see in the implementation with something better.

The current CoreCLR plumbing won't take advantage of the intrinsics attribute for this case; but it might in the future and there are other places where we have done this.